examples/benchmarks/other/nbe.effekt

```diff
@@ -0,0 +1,156 @@
+  case VNeu(NVar(n)) => Var(n)
+  case VNeu(NApp(a, b)) => App(reify(VNeu(a)), reify(b))
+  case _ => {
+    val n = do fresh()
+    Abs(n, reify(apply(v, reflect(n))))
+  }
+}
+
+/// strong normalization of the term
+def normalize(t: Term): Term = {
+  var i = 0
+  try reify(eval(empty[Name, Value](compareInt), t))
+  with fresh { i = i + 1; resume(i) }
+}
+
+/// parse named term from BLC stream (de Bruijn)
+def parse(): Term / { read[Bool], stop } = {
+  def go(): Term / { fresh, lookup } =
+    (do read[Bool](), do read[Bool]) match {
+      case (false, false) => {
+        val n = do fresh()
+        Abs(n, try go() with lookup { i =>
+          resume(if (i == 0) n else do lookup(i - 1))
+        })
+      }
+      case (false, true ) => App(go(), go())
+      case (true , false) => Var(do lookup(0))
+      case (true , true ) => {
+        var i = 1
+        while (do read[Bool]()) i = i + 1
+        Var(do lookup(i))
+      }
+    }
+
+  var i = 0
+  try go()
+  with fresh { i = i + 1; resume(i) }
+  with lookup { n =>
+    println("error: free variable " ++ n.show)
+    exit(1)
+  }
+}
+
+/// helper function for pretty string interpolation of terms
+def pretty { s: () => Unit / { literal, splice[Term] } }: String = {
+  with stringBuffer
+
+  try { s(); do flush() }
+  with literal { l => resume(do write(l)) }
+  with splice[Term] { t =>
+    t match {
+      case Abs(n, t) => do write("λ" ++ n.show ++ pretty".${t}")
+      case App(a, b) => do write(pretty"(${a} ${b})")
+      case Var(v)    => do write(v.show)
+    }
+    resume(())
+  }
+}
+
+/// convert char stream to bit stream, skipping non-bit chars
+def bits { p: () => Unit / { read[Bool], stop } }: Unit / read[Char] =
+  try exhaustively { p() }
+  with read[Bool] {
+    with exhaustively
+    val c = do read[Char]()
+    if (c == '0') resume { false }
+    if (c == '1') resume { true  }
+  }
+
+/// evaluate the input BLC string and compare it to its prettified normal form
+def normalizes?(input: String, normal: String) = {
+  with feed(input)
+  with bits
+  assert(pretty"${normalize(parse())}", normal)
+}
+
+def main() = mainSuite("Lambda NbE") {
+  test("S ~> S") {
+    val t = "00000001011110100111010"
+    normalizes?(t, "λ1.λ2.λ3.((1 3) (2 3))")
+  }
+
+  test("(ι (ι (ι (ι ι)))) ~> S") {
+    val t = "010001011000000001011110100111010000011001000101100000000101111010011101000001100100010110000000010111101001110100000110010001011000000001011110100111010000011000010110000000010111101001110100000110"
+    normalizes?(t, "λ1.λ2.λ3.((1 3) (2 3))")
+  }
+
+  test("(2^3)%3 == 2 // using Church numerals") {
+    val t = "010101000000010101100000011100001011100000011100101111011010100100000110110000010000101011110111100101011100000000111100001011110101100010000001101100101011100000110001000100000011100111010000001110011100111010000001110011100111010"
+    normalizes?(t, "λ1.λ2.(1 (1 2))")
+  }
+
+  // TODO: is there an off-by-one in the string buffer?
```

Comment:
This is weird. Either I'm interpreting the test error message wrong, or the `pretty` string interpolation forgets to write the *last* closing parenthesis. Could this be an off-by-one error in the stringbuffer flushing/growing?